hollow/src/main/java/com/netflix/hollow/api/consumer/HollowConsumer.java

```diff
@@ -1102,13 +1102,20 @@ public B withRefreshListeners(HollowConsumer.RefreshListener... refreshListeners
          * @return this builder
          * @throws IllegalArgumentException if provided API class is {@code HollowAPI} instead of a subclass
          */
-        public B withGeneratedAPIClass(Class<? extends HollowAPI> generatedAPIClass) {
+        public B withGeneratedAPIClass(Class<? extends HollowAPI> generatedAPIClass, String cachedType, String... additionalCachedTypes) {
             if (HollowAPI.class.equals(generatedAPIClass))
                 throw new IllegalArgumentException("must provide a code generated API class");
-            this.apiFactory = new HollowAPIFactory.ForGeneratedAPI<>(generatedAPIClass);
+            String[] types = new String[additionalCachedTypes.length + 1];
+            types[0] = cachedType;
+            System.arraycopy(additionalCachedTypes, 0, types, 1, additionalCachedTypes.length);
+            this.apiFactory = new HollowAPIFactory.ForGeneratedAPI<>(generatedAPIClass, types);
             return (B)this;
         }
 
+        public B withGeneratedAPIClass(Class<? extends HollowAPI> generatedAPIClass) {
+            return withGeneratedAPIClass(generatedAPIClass, null);
```

Comment:
This is in turn leading to an array of one element equal to `null` being passed to the underlying `HollowAPIFactory`. I think the other comment regarding handling of null values in the cached types will fix this, or we could also just retain the original implementation of this method